src/averageAges.js

```diff
@@ -55,7 +74,22 @@ function calculateWomenAverageAge(people, withChildren) {
  * @return {number}
  */
 function calculateAverageAgeDiff(people, onlyWithSon) {
-  // write code here
+  const womenWithSon = onlyWithSon
+    ? people
+      .filter(person => people.some(mother =>
+        mother.name === person.mother) && person.sex === 'm')
+    : people
+      .filter(person => people.some(mother =>
+        mother.name === person.mother));
```

Comment:
Also, you can do like that
```suggestion
  const womenWithSon =  people.filter(person => onlyWithSon
    ? hasMother(people, person.mother) && person.sex === 'm')
    : hasMother(people,person.mother);
```